rust/src/daemon.rs

```diff
@@ -131,6 +149,139 @@ fn deployment_populate_variant_origin(
     Ok(())
 }
 
+pub(crate) fn write_message(conn: &OwnedFd, message: SocketMessage) -> Result<()> {
```

Comment:
I think now that we're using a stream, we probably need to change all this so that we write exactly one message and then close the connection.  And relatedly, instead of using `send` we need to `write_all`.

Reply:
Does this mean we send a `rustix::net::shutdown` from the daemon after we have done acknowledgment?